### PR TITLE
mocked api and docker

### PR DESCRIPTION
**Pull request by fcallejon**

### CODE REVIEW

**Comment by fcallejon** [comment]
👍

File: api/src/endpoints/postJoin.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+const users = mongoose.model("users", usersSchema);
+
+const postJoin = {
+  method: "POST",
+  url: "/api/join/:userId",
+  schema: {
+    body: {
+      type: "object",
+      properties: {
+        id: { type: "string" },
```
Comment (agrcrobles) [inline api/src/endpoints/postJoin.js:13]: change to number

File: api/src/endpoints/postJoin.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+      const user = await users.findById(request.params.userId);
+      if (!user) {
+        console.error(`User ${request.params.userId} doesn't exists.`);
+        reply.code(400).send("Invalid Data");
+        return;
+      }
+
+      if (
+        node.u.filter(u => u._id.toString() === user._id.toString()).length > 0
```
Comment (agrcrobles) [inline api/src/endpoints/postJoin.js:46]: ?

File: api/src/endpoints/postSignin.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        te: signingUser.te,
+        o: signingUser.o
+      }
+    });
+
+    try {
+      const savedUser = await user.save();
+      reply.code(200).send(savedUser._id);
+    } catch (e) {
+      if (e.name === "BulkWriteError" && e.code === 11000) {
```
Comment (agrcrobles) [inline api/src/endpoints/postSignin.js]: didn't work for my db

File: package.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  "scripts": {
+    "server-dev": "./node_modules/nodemon/bin/nodemon.js ./api/src/index.js",
+    "test": "./node_modules/tap/bin/run.js './{,!(node_modules)/**}/*.test.js'",
+    "start": "node ./api/src",
+    "docker-build": "./scripts/build.sh",
+    "docker-start": "./scripts/start.sh"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged",
```
Comment (agrcrobles) [inline package.json]: doesn't work

File: api/src/endpoints/getMyMasternodes.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      reply.type("application/json").code(200);
+      const user = await users.findById(request.params.userId);
+      if (!user || (user.ok && !user.ok)) {
+        console.error(`User with id '${request.params.userId}' not found.`);
+        reply.code(400).send();
+        return;
+      }
+      const joinedNodes = await nodes
+        .find({ "u._id": user._id })
+        .select({ _id: 1, n: 1, d: 1, f: 1 });
```
Comment (agrcrobles) [inline api/src/endpoints/getMyMasternodes.js:20]: _id should be autogenerated

**Review by agrcrobles — CHANGES_REQUESTED** [review]
added comments

File: api/src/endpoints/postJoin.js (hunk 1 above)
Comment (fcallejon) [reply api/src/endpoints/postJoin.js:13]: Actually we can't. The type it's ObjectId.

File: api/src/endpoints/postJoin.js (hunk 2 above)
Comment (fcallejon) [reply api/src/endpoints/postJoin.js:46]: Checking you aren't in the node already.

File: api/src/endpoints/postSignin.js (hunk 3 above)
Comment (fcallejon) [reply api/src/endpoints/postSignin.js]: Need to add the script for the unique index.

File: package.json (hunk 4 above)
Comment (fcallejon) [reply package.json]: Yep, lint-staged it's using eslint not prettier.

File: api/src/endpoints/getMyMasternodes.js (hunk 5 above)
Comment (fcallejon) [reply api/src/endpoints/getMyMasternodes.js:20]: Id it's autogenerated. Don't get what you mean.

Comment (fcallejon) [reply api/src/endpoints/getMyMasternodes.js:20]: ```Select``` just build a projection.

File: package.json (hunk 4 above)
Comment (fcallejon) [reply package.json]: Removed all together as we are using vscode's ```"editor.formatOnSave": true``` anyways.

File: api/src/endpoints/postSignin.js (hunk 3 above)
Comment (fcallejon) [reply api/src/endpoints/postSignin.js]: Changed the way we check instead of relaying on unique-index

**Comment by fcallejon** [comment]
Please check the updates.

File: api/src/endpoints/postJoin.js (hunk 2 above)
Comment (agrcrobles) [reply api/src/endpoints/postJoin.js:46]: usa exists gato

**Review by agrcrobles — APPROVED** [review]

File: api/src/endpoints/postJoin.js (hunk 2 above)
Comment (fcallejon) [reply api/src/endpoints/postJoin.js:46]: Isn't $exists to check for a field to exist?

**Pull request merged**